test/test_util_pipe.py

```diff
@@ -78,19 +78,25 @@ def test_pipe_protocol(self):
             pipe_writer_handler = pipe.Pipe(self.NAME, self.VERSION)
 
         __pool = concurrent.futures.ThreadPoolExecutor()
+        for count in range(100):
+            for message in to_send:
+                for iteration in range(20):
+                    log.debug("No. %d attempt to send: %r", iteration+1, message)
+                    plistener = __pool.submit(pipe_listener, pipe_listener_handler)
+                    pwriter = __pool.submit(pipe_writer, pipe_writer_handler, message)
+                    to_break = False
+                    try:
+                        self.assertEqual(plistener.result(timeout=6.5), message,
```

Comment:
Having `assertEqual()` in a try block means in some case the test will pass without even checking correct behavior.

If we expect an exception, then we need to check it happens.

Reply:
So, basically, `plistener.result(timeout=6.5)` should be placed in a try-except block to handle timeout errors, while assertequal should be in the normal loop, right?

Reply:
Yes, the test should always end on an assertion (or be explicitly skipped). Sometimes an exception is expected, but in this case we still assert for it.